File: CursoLogica/Linguagens/Python/Atividades/diagonal_negativos.py
ordem = int(input("Qual a ordem da matriz? "))

matriz = [[0 for x in range(ordem)] for x in range(ordem)]

for i in range(ordem):
    for j in range(ordem):
        matriz[i][j] = int(input(f"Elemento [{i},{j}]: "))

negativos = 0
for i in range(ordem):
    for j in range(ordem):
        if matriz[i][j] < 0:
            negativos += 1

print("DIAGONAL PRINCIPAL:")
for i in range(ordem):
    print(matriz[i][i], end=" ")

print()
print(f"QUANTIDADE DE NEGATIVOS = {negativos}")

File: CursoLogica/Linguagens/C++/Atividades/diagonal_negativos.cpp
#include <iomanip>
#include <iostream>

using namespace std;

int main() {

    int ordem, i, j, qntNegativos;

    cout << "Qual a ordem da matriz? ";
    cin >> ordem;

    int matriz[ordem][ordem];

    for (i = 0; i < ordem; i++) {
        for (j = 0; j < ordem; j++) {
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> matriz[i][j];
        }
    }

    qntNegativos = 0;
    for (i = 0; i < ordem; i++) {
        for (j = 0; j < ordem; j++) {
            if (matriz[i][j] < 0) {
                qntNegativos = qntNegativos + 1;
            }
        }
    }

    cout << "DIAGONAL PRINCIPAL:\n";
    for (i = 0; i < ordem; i++) {
        cout << matriz[i][i] << " ";
    }

    cout << "\nQUANTIDADE DE NEGATIVOS = " << qntNegativos;

    return 0;
}
